lib/AST/Decl.cpp

```diff
@@ -2061,9 +2061,18 @@ PatternBindingDecl::create(ASTContext &Ctx, SourceLoc StaticLoc,
     // FIXME: We ought to reconsider this since it won't recontextualize any
     // closures/decls present in the initialization expr. This currently should
     // only affect implicit code though.
-    if (!initContext && !Parent->isLocalContext())
+    if (!initContext && !Parent->isLocalContext()) {
       initContext = PatternBindingInitializer::create(Parent);
 
+    }
+
+    if (auto firstVar = PBD->getAnchoringVarDecl(idx)) {
+      if (auto attributeInit =
+              firstVar->getAttrs().findCustomAttributeInitializer()) {
+        attributeInit->setEnclosingInitializer(initContext);
```

Comment:
Not that each custom attribute has its own initializer context. so we need to contextualize them all, not only the first one.

Reply:
Argh! You're right, of course. Thank you!